util/timer_test.cc

```diff
@@ -0,0 +1,240 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include "util/timer.h"
+
+#include "db/db_test_util.h"
+
+class TimerTest : public testing::Test {
+ public:
+  TimerTest()
+      : mock_env_(new ROCKSDB_NAMESPACE::MockTimeEnv(
+          ROCKSDB_NAMESPACE::Env::Default())) {}
+
+ protected:
+  std::unique_ptr<ROCKSDB_NAMESPACE::MockTimeEnv> mock_env_;
+};
+
+TEST_F(TimerTest, SingleScheduleOnceTest) {
+  const uint64_t kSecond = 1000000;  // 1sec = 1000000us
+  const int kIterations = 1;
+  uint64_t time_counter = 0;
+  mock_env_->set_current_time(0);
+  ROCKSDB_NAMESPACE::port::Mutex mutex;
+  ROCKSDB_NAMESPACE::port::CondVar test_cv(&mutex);
+
+  ROCKSDB_NAMESPACE::Timer timer(mock_env_.get());
```

Comment:
Nit: the test code is all inside namespace rocksdb, so no need to add "ROCKSDB_NAMESPACE::" which may scare readers with the macro.